### ISSUE TITLE
Implement clippable text

### ISSUE THREAD

**Issue by ajyoon** [ajyoon/brown#9]
Paths can currently be clipped arbitrarily for seamless flowable rendering. There are many use cases where it would be useful to have clippable text. Particularly, many specially shaped lines can be implemented with text. For example, SMuFL includes random wiggle lines which can be laid out as text to draw squiggles. A long enough line would ideally be able to wrap seamlessly around flowable lines, but currently no such thing is supported.

```python
random_wiggles = [random.choice(['wiggleRandom1',
                                 'wiggleRandom2',
                                 'wiggleRandom3',
                                 'wiggleRandom4']) for i in range(150)]

MusicTextObject((Mm(25), lower_staff.unit(4)),
                random_wiggles,
                lower_staff).render()
```

![screenshot from 2017-02-08 00-05-52](https://cloud.githubusercontent.com/assets/15976214/22724181/91020770-ed92-11e6-822c-1ef70a2df5fd.png)

This should probably be implemented simply by extending `QEnhancedTextItem` with logic mirroring the implementation in `QClippingPath`

**Comment by ajyoon** [comment]
Progress in 26f2d041104e2a1171acc00f70a500bf41e8328f

Still need to fix some strange bug with clipping scaled text...

**Comment by ajyoon** [comment]
Fixed & finished in 0bbb15527b60625ca74084a3ca1ab3922ef9f8bb
